server/src/test/java/org/opensearch/index/codec/composite912/datacube/startree/StarTreeKeywordDocValuesFormatTests.java

```diff
@@ -0,0 +1,565 @@
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeTestUtils;
+import org.opensearch.index.compositeindex.datacube.startree.index.StarTreeValues;
+import org.opensearch.index.compositeindex.datacube.startree.utils.iterator.SortedSetStarTreeValuesIterator;
+import org.opensearch.index.mapper.NumberFieldMapper;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.index.compositeindex.CompositeIndexConstants.STAR_TREE_DOCS_COUNT;
+import static org.opensearch.index.compositeindex.datacube.startree.StarTreeTestUtils.assertStarTreeDocuments;
+
+/**
+ * Star tree doc values Lucene tests
+ */
+@LuceneTestCase.SuppressSysoutChecks(bugUrl = "we log a lot on purpose")
+public class StarTreeKeywordDocValuesFormatTests extends AbstractStarTreeDVFormatTests {
+
+    public StarTreeKeywordDocValuesFormatTests(StarTreeFieldConfiguration.StarTreeBuildMode buildMode) {
+        super(buildMode);
+    }
+
+    public void testStarTreeKeywordDocValues() throws IOException {
+        Directory directory = newDirectory();
+        IndexWriterConfig conf = newIndexWriterConfig(null);
+        conf.setMergePolicy(newLogMergePolicy());
+        RandomIndexWriter iw = new RandomIndexWriter(random(), directory, conf);
+        Document doc = new Document();
+        doc.add(new StringField("_id", "1", Field.Store.NO));
+        doc.add(new SortedNumericDocValuesField("sndv", 1));
+        doc.add(new SortedSetDocValuesField("keyword1", new BytesRef("text1")));
+        doc.add(new SortedSetDocValuesField("keyword2", new BytesRef("text2")));
+        iw.addDocument(doc);
+        doc = new Document();
+        doc.add(new StringField("_id", "2", Field.Store.NO));
+        doc.add(new SortedNumericDocValuesField("sndv", 1));
+        doc.add(new SortedSetDocValuesField("keyword1", new BytesRef("text11")));
+        doc.add(new SortedSetDocValuesField("keyword2", new BytesRef("text22")));
+        iw.addDocument(doc);
+        iw.flush();
+        iw.deleteDocuments(new Term("_id", "2"));
+        iw.flush();
+        doc = new Document();
+        doc.add(new StringField("_id", "3", Field.Store.NO));
+        doc.add(new SortedNumericDocValuesField("sndv", 2));
+        doc.add(new SortedSetDocValuesField("keyword1", new BytesRef("text1")));
+        doc.add(new SortedSetDocValuesField("keyword2", new BytesRef("text2")));
+        iw.addDocument(doc);
+        doc = new Document();
+        doc.add(new StringField("_id", "4", Field.Store.NO));
+        doc.add(new SortedNumericDocValuesField("sndv", 2));
+        doc.add(new SortedSetDocValuesField("keyword1", new BytesRef("text11")));
+        doc.add(new SortedSetDocValuesField("keyword2", new BytesRef("text22")));
+        iw.addDocument(doc);
+        iw.flush();
+        iw.deleteDocuments(new Term("_id", "4"));
```

Comment:
What are we trying to test with doc deletes?

Reply:
I was trying to simulate the segDocValuesProducer with multiple DVPs case, even with soft deletes set in config, not able to replicate that scenario

Reply:
addressed this thanks for the help